desc/__init__.py

```diff
@@ -127,11 +127,15 @@ def set_device(kind="cpu"):
             set_device(kind="cpu")
             return
         devices = [dev for dev in devices if dev["index"] in gpu_ids]
-        for dev in devices:
-            mem = dev["mem_total"] - dev["mem_used"]
-            if mem > maxmem:
-                maxmem = mem
-                selected_gpu = dev
+
+        if (not (gpuid == None)) and (str(gpuid) in gpu_ids):
```

Comment:
```suggestion
        if gpuid is not None and (str(gpuid) in gpu_ids):
```

Reply:
Updates are made according to suggestion.